### ISSUE TITLE
HTTP/2 protocol was not selected. (nghttp2 expects h2)

### ISSUE THREAD

**Issue by s0j0urn** [nghttp2/nghttp2#578]
I am running Jettyv9.3.8 on RaspberryPi_1.  I have enabled  http/2, ALPN, SSL modules in jetty.

To this RaspberryPi_1, I have another raspberryPI_2 and my laptop connected.
When I use nghttp or h2load on RaspberryPi_2 to connect to jetty server running on RaspberryPi_1 I am able to connect. 

However, when I run nghttp or h2load on my laptop to connect to jetty server running on RaspberryPi_1. I start getting protocol negotiation errors.

Laptop to Jetty(on RaspberryPI_1):
```

nghttp -nvs https://10.42.0.31:8443/my-gallery/index.html
[  0.001] Connected
[ERROR] HTTP/2 protocol was not selected. (nghttp2 expects h2)
Some requests were not processed. total=1, processed=0
***** Statistics *****

Request timing:
  responseEnd: the  time  when  last  byte of  response  was  received
               relative to connectEnd
 requestStart: the time  just before  first byte  of request  was sent
               relative  to connectEnd.   If  '*' is  shown, this  was
               pushed by server.
      process: responseEnd - requestStart
         code: HTTP status code
         size: number  of  bytes  received as  response  body  without
               inflation.
          URI: request URI

see http://www.w3.org/TR/resource-timing/#processing-model

sorted by 'complete'
```

From RaspberryPi_2 to Jetty(RaspberryPi_1)

```
nghttp -nvs https://x.x.x.x:8443/my-galleryindex.html

[  0.006] Connected
The negotiated protocol: h2
[  0.817] send SETTINGS frame <length=12, flags=0x00, stream_id=0>
          (niv=2)
          [SETTINGS_MAX_CONCURRENT_STREAMS(0x03):100]
          [SETTINGS_INITIAL_WINDOW_SIZE(0x04):65535]
[  0.818] send PRIORITY frame <length=5, flags=0x00, stream_id=3>
          (dep_stream_id=0, weight=201, exclusive=0)
.............
..............
..............
.............
[  0.831] recv (stream_id=13) :status: 200
[  0.831] recv (stream_id=13) server: Jetty(9.3.8.v20160314)
[  0.831] recv (stream_id=13) last-modified: Sun, 03 Apr 2016 16:06:46 GMT
[  0.831] recv (stream_id=13) content-length: 728
[  0.832] recv (stream_id=13) content-type: text/html
[  0.832] recv (stream_id=13) accept-ranges: bytes
[  0.832] recv HEADERS frame <length=66, flags=0x04, stream_id=13>
          ; END_HEADERS
          (padlen=0)
          ; First response header
[  0.833] recv DATA frame <length=728, flags=0x01, stream_id=13>
          ; END_STREAM
[  0.834] send GOAWAY frame <length=8, flags=0x00, stream_id=0>
          (last_stream_id=0, error_code=NO_ERROR(0x00), opaque_data(0)=[])
***** Statistics *****

Request timing:
  responseEnd: the  time  when  last  byte of  response  was  received
               relative to connectEnd
 requestStart: the time  just before  first byte  of request  was sent
               relative  to connectEnd.   If  '*' is  shown, this  was
               pushed by server.
      process: responseEnd - requestStart
         code: HTTP status code
         size: number  of  bytes  received as  response  body  without
               inflation.
          URI: request URI

see http://www.w3.org/TR/resource-timing/#processing-model

sorted by 'complete'

id  responseEnd requestStart  process code size request path
 13    +16.96ms      +1.91ms  15.04ms  200  728 /my-gallery/index.html
```

I am running version `nghttp nghttp2/1.8.0` on my laptop as well as RaspberryPi_2.

**Comment by tatsuhiro-t** [comment]
As far as I know, jetty only supports ALPN to negotiate application protocol.
If your laptop has openssl 1.0.1, it has NPN, but it does not support ALPN.
nghttp uses NPN to negotiate protocol, but jetty does not support it.  So no negotiation happens, and you get that error.

**Comment by s0j0urn** [comment]
Yes, it turned out to be openssl version error.

For someone who comes across the similar problem in future:
I had Anaconda Python installed on my laptop. It was returning the openssl version as `1.0.2g March 2016` . I removed Anaconda python completely as it was not my immediate need and then checked the openssl version which turned out to be `1.0.1`. I then installed `openssl 1.0.2f` and made sure `libcrypto.a` and` libssl.a` libraries are on the correct path.

Sorry for the unnecessary trouble @tatsuhiro-t . 
Thank you.
